plugins/exporter.koplugin/clip.lua

```diff
@@ -245,12 +245,9 @@ function MyClipping:parseHighlight(highlights, bookmarks, book)
             clipping.text = self:getText(item.text)
             clipping.chapter = item.chapter
             for _, bookmark in pairs(bookmarks) do
-                if bookmark.datetime == item.datetime and bookmark.text then
-                    local tmp = string.gsub(bookmark.text, "^Page %[?%d*%]?%d+ ", "")
-                    local bookmark_quote = string.gsub(tmp, " @ %d%d%d%d%-%d%d%-%d%d %d%d:%d%d:%d%d", "")
-                    if bookmark_quote ~= clipping.text then
-                        clipping.note = bookmark_quote
-                    end
+                if bookmark.datetime == item.datetime and bookmark.text and
+                        not self.ui.bookmark:isBookmarkAutoText(bookmark) then
```

Comment:
`./luajit: plugins/exporter.koplugin/clip.lua:249: attempt to index field 'ui' (a nil value)`

Reply:
Hm. `isBookmarkAutoText` actually might not work well here when called from "Export all notes in your library", which I'm guessing is what you tried here. I'm looking into alternatives.

Reply:
This way:
```lua
local ReaderUI = require("apps/reader/readerui")
```
and
```lua
not ReaderUI.instance.bookmark:isBookmarkAutoText(bookmark) then
```

Reply:
That didn't work for me. At least not if I select "Export all notes in this book" while no document is open. Were you selecting "Export all notes in this book" while a document is open?

```
./luajit: plugins/exporter.koplugin/clip.lua:250: attempt to index field 'instance' (a nil value)
stack traceback:
        plugins/exporter.koplugin/clip.lua:250: in function 'parseHighlight'
        plugins/exporter.koplugin/clip.lua:300: in function 'parseHistoryFile'
        plugins/exporter.koplugin/clip.lua:313: in function 'parseHistory'
        plugins/exporter.koplugin/main.lua:468: in function 'exportAllNotes'
        plugins/exporter.koplugin/main.lua:319: in function 'action'
        frontend/ui/uimanager.lua:1201: in function '_checkTasks'
        frontend/ui/uimanager.lua:1622: in function 'handleInput'
        frontend/ui/uimanager.lua:1730: in function 'run'
        ./reader.lua:323: in main chunk
        [C]: at 0x0106e168c0
```

Reply:
I am afraid isBookmarkAutoText is a good solution for the current document only.
Maybe return to some strings gsub and check, in massive export? With translated _("Page").

Reply:
Besides users editing that string or writing something similar, if you switch between languages you'll presumably have a variety of different translated strings for "Page" as well.

Reply:
IMO, the issue with changing language after previously taking notes is a more general flaw with the auto-text concept, and an issue that also appears elsewhere in koreader. My personal preference would be to get rid of auto text and provide the user with some new mechanism of inserting the quoted text if they want to edit it. I don't think I'm going to be able to fix that well here since it has broader implications, but it's something I'll look at separately. I'm open to opinions if anyone has ideas for more targeted changes though.

I just pushed an update that uses an improved and translated pattern. It manages to keep the original behavior of allowing modifying the quoted text and extracting it. Otherwise, it falls back to just using the full string.

Reply:
> to get rid of auto text

Please look at
https://github.com/koreader/koreader/pull/8301#issuecomment-934444178
https://github.com/koreader/koreader/pull/8301#issuecomment-939061663

Reply:
Thanks for the context. I'll have to read up more on the improvements being discussed for bookmarks. I'm glad that others also want to improve the weirdness of auto-text. :)

I also noticed from that discussion that I needed a change to support the "Add page number/timestamp to bookmark" setting, which I just pushed.

Any remaining concerns for how I'm handling things for now though?